pkg/api/query/v1.go

```diff
@@ -806,11 +815,15 @@ func NewExemplarsHandler(client exemplars.UnaryClient, enablePartialResponse boo
 			Query:                   r.FormValue("query"),
 			PartialResponseStrategy: ps,
 		}
-		exemplarsData, warnings, err := client.Exemplars(r.Context(), req)
-		if err != nil {
-			return nil, nil, &api.ApiError{Typ: api.ErrorInternal, Err: errors.Wrap(err, "retrieving exemplars")}
+
+		tracing.DoInSpan(ctx, "retrieve_exemplars", func(ctx context.Context) {
+			exemplarsData, exemplarWarnings, exemplarError = client.Exemplars(r.Context(), req)
```

Comment:
```suggestion
			exemplarsData, exemplarWarnings, exemplarError = client.Exemplars(ctx, req)
```
To propagate correct context, we should use the one in closure.

Reply:
good catch :+1: